File: csharp/src/calisthenics-exercises/OneIndentationLevel.cs
namespace RefactoringAvanzado.CalisthenicExercises;

public class OrderItem
{
    public double Price { get; set; }

    public OrderItem(double price)
    {
        Price = price;
    }
}

public class OrderCustomer
{
    public bool IsVip { get; set; }

    public OrderCustomer(bool isVip)
    {
        IsVip = isVip;
    }
}

public class Order
{
    public string Id { get; set; }
    public List<OrderItem> Items { get; set; }
    public OrderCustomer Customer { get; set; }

    public Order(string id, List<OrderItem> items, OrderCustomer customer)
    {
        Id = id;
        Items = items;
        Customer = customer;
    }
}

public static class OrderProcessor
{
    public static double ProcessOrdersWithDiscounts(List<Order> orders)
    {
        double total = 0;
        foreach (var order in orders)
        {
            if (order.Items != null && order.Items.Count > 0)
            {
                foreach (var item in order.Items)
                {
                    if (order.Customer != null && order.Customer.IsVip)
                    {
                        if (item.Price > 100)
                        {
                            total += item.Price * 0.8;
                        }
                        else
                        {
                            total += item.Price * 0.9;
                        }
                    }
                    else
                    {
                        if (item.Price > 100)
                        {
                            total += item.Price * 0.95;
                        }
                        else
                        {
                            total += item.Price;
                        }
                    }
                }
            }
        }
        return total;
    }
}

File: php/src/calisthenics-exercises/OneIndentationLevel.php
<?php

namespace Calisthenics;

class OneIndentationLevel
{
    /**
     * @param array<array{id: string, items: array<array{price: float}>, customer: array{isVip: bool}}> $orders
     */
    public static function processOrdersWithDiscounts(array $orders): float
    {
        $total = 0;
        foreach ($orders as $order) {
            if (isset($order['items']) && count($order['items']) > 0) {
                foreach ($order['items'] as $item) {
                    if (isset($order['customer']) && $order['customer']['isVip']) {
                        if ($item['price'] > 100) {
                            $total += $item['price'] * 0.8; // gran descuento VIP
                        } else {
                            $total += $item['price'] * 0.9; // pequeño descuento VIP
                        }
                    } else {
                        if ($item['price'] > 100) {
                            $total += $item['price'] * 0.95; // gran descuento regular
                        } else {
                            $total += $item['price']; // sin descuento
                        }
                    }
                }
            }
        }
        return $total;
    }
}
